lib/staging.js

```diff
@@ -0,0 +1,457 @@
+    super(cli, dir);
+    if (!this.config.branch) {
+      this.warnForMissing();
+    }
+
+    this.cli = cli;
+    this.dir = dir;
+    this.paginate = paginate;
+    this.skipGH = skipGH;
+    this.releaseLine = releaseLine;
+    this.reporter = reporter;
+    this.seenRequestBackportIds = new Set();
+  }
+
+  get isLTS() {
+    if (this.#isLTS !== undefined) {
+      return this.#isLTS;
+    }
+    const header = readFile(path.resolve(this.dir, 'src', 'node_version.h'));
+    this.#isLTS = header.indexOf('#define NODE_VERSION_IS_LTS 1') > -1;
+    return this.isLTS;
+  }
+
+  get ncuDir() {
+    return getNcuDir(this.dir);
+  }
+
+  get stateDir() {
+    return path.resolve(this.ncuDir, 'staging');
+  }
+
+  get branchDiffCache() {
+    return path.resolve(this.ncuDir, 'branch-diff-output' + this.stagingBranch);
+  }
+
+  get state() {
+    return readJson(this.stateDir);
+  }
+
+  set state(obj) {
+    writeJson(this.stateDir, obj);
+  }
+
+  // If the release line is not yet set, tries to retrieve it from the
+  // current ncu set branch, if that fails to work, prompt the user to set it
+  async maybeSetReleaseLine() {
+    if (!this.releaseLine || !validNumber(parseInt(this.releaseLine, 10))) {
+      const majorVersion = this.config.branch.split('.x-staging')[0].slice(1);
+      if (!majorVersion || !validNumber(parseInt(majorVersion, 10))) {
+        const promptReleaseLine = await this.cli.prompt(
+          'Enter the major version for the target release line:',
+          { questionType: 'input', noSeparator: true, defaultAnswer: '' }
+        );
+
+        if (!promptReleaseLine || !validNumber(parseInt(promptReleaseLine, 10))) {
+          this.cli.error('A release line is required to port commits to.');
+          return;
+        } else {
+          this.#setReleaseLine(promptReleaseLine);
+        }
+      }
+      this.#setReleaseLine(majorVersion);
+    }
+  }
+
+  async getBranchDiff() {
+    const cli = this.cli;
+    const upstream = this.config.upstream;
+    const majorVersion = this.releaseLine;
+
+    const excludeLabels = [
+      'semver-major',
+      `dont-land-on-v${majorVersion}.x`,
+      `backport-requested-v${majorVersion}.x`,
+      `backported-to-v${majorVersion}.x`,
+      `backport-blocked-v${majorVersion}.x`,
+      `backport-open-v${majorVersion}.x`
+    ];
+
+    let comparisonBranch;
+    if (this.isLTS) {
+      excludeLabels.push('baking-for-lts');
+      const res = await fetch('https://nodejs.org/dist/index.json');
+      if (!res.ok) throw new Error('Failed to fetch', { cause: res });
+      const [latest] = await res.json();
+      // Assume Current branch matches tag with highest semver value.
+      const latestReleaseLine = semver.coerce(latest.version).major;
+      if (!validNumber(latestReleaseLine)) {
+        throw new Error('Could not determine latest release line');
+      }
+      comparisonBranch = `v${latestReleaseLine}.x`;
```

Comment:
The latest current may not be the best thing to compare to. For example, if you were to take the latest v23.x today, that would be v23.3.0 which was only released one week ago and presumably commits would be identified that having spent the usual "two weeks in a current release" (depending, of course, on when the planned date for the release being prepared is).